File: Sources/ShoppingList.Web/ClientApp/src/app/models/shopping-list.model.ts
import { Expose, Type } from 'class-transformer';
import { ShoppingItemModel } from './shopping-item.model';

export class ShoppingListModel {
    @Expose() id!: string;
    @Expose() title!: string;

    @Expose()
    @Type(() => ShoppingItemModel)
    items!: ShoppingItemModel[];
}

File: Sources/ShoppingList.Logic/Models/ShoppingListModel.cs
using System.Collections.Generic;

namespace ShoppingList.Logic.Models
{
	public class ShoppingListModel
	{
		public IdModel Id { get; set; }

		public string Title { get; set; }

		public IReadOnlyCollection<ShoppingItemModel> Items { get; set; }
	}
}
